### PR TITLE
[OC-5241] Revert ecommerce theme customizations

### PR DESCRIPTION
**Pull request by lgp171188**
These have been moved into the theme. This is essentially a reverse patch for the changes between the upstream `open-release/ginkgo.master` branch and this repository's `opencraft-release/ginkgo.master-hms` branch.

**Testing instructions**:
* Make a diff between the upstream `open-release/ginkgo.master` branch and the source branch of this PR only for the `ecommerce/static/` and `ecommerce/templates` directory.
* Verify that the changes in this PR are reverting the above diff exactly.

I have attached the [diff](https://github.com/open-craft/ecommerce/files/2395564/c.diff.txt) and the [list of files changed](https://github.com/open-craft/ecommerce/files/2395565/c.txt) for easy reference.

### CODE REVIEW

**Review by UmanShahzad — COMMENTED** [review]
This looks good; I ran the following and got no difference:

```
$ git diff open-release/ginkgo.2 oc/guruprasad/revert-ecommerce-theme-customizations -- ecommerce/static/ ecommerce/templates/
```

I need to look at a few more things but will approve after.

File: ecommerce/templates/customer/emails/commtype_credit_receipt_subject.txt
Comment (UmanShahzad) [inline ecommerce/templates/customer/emails/commtype_credit_receipt_subject.txt:2]: @lgp171188 This override is missing in the theme repo.

File: ecommerce/templates/edx/partials/_student_navbar.html
Comment (UmanShahzad) [inline ecommerce/templates/edx/partials/_student_navbar.html:9]: For the lines below this I saw that you included them in the theme. Do we have to include them if we only have changes in the `navbar_brand` block?

File: ecommerce/templates/oscar/basket/basket.html
Comment (UmanShahzad) [inline ecommerce/templates/oscar/basket/basket.html:12]: Similar comment here as above; I saw that you are overriding the entire file basically rather than just the block - is that necessary?

File: ecommerce/templates/oscar/basket/partials/hosted_checkout_basket.html
Comment (UmanShahzad) [inline ecommerce/templates/oscar/basket/partials/hosted_checkout_basket.html:21]: Similar comment for this file - there are a lot of named blocks available that we can override alone rather than the entire file, but I'm not sure if named template block overrides actually work the way they do in the edx-platform.

File: ecommerce/static/images/logo_primary.svg
Comment (UmanShahzad) [inline ecommerce/static/images/logo_primary.svg:1]: I couldn't comment on the above, but all these custom fonts and images seem to be missing from the theme repo.

File: ecommerce/static/js/views/coupon_form_view.js
Comment (UmanShahzad) [inline ecommerce/static/js/views/coupon_form_view.js:375]: Wow, we had to override this entire huge file just to get this? I'm really wondering if there's a better way to change this value, e.g. through some setting. But since Hawthorn's already here that may not work out this time around.

File: ecommerce/static/images/logo_primary.svg
Comment (lgp171188) [reply ecommerce/static/images/logo_primary.svg:1]: @UmanShahzad, I have added them in the theme repository now.

File: ecommerce/templates/edx/partials/_student_navbar.html
Comment (lgp171188) [reply ecommerce/templates/edx/partials/_student_navbar.html:9]: @UmanShahzad, based on my understanding of ecommerce theming (and Django template language), the template in the theme is picked instead of the default in the ecommerce repository. If the default template is not a full template (in this case it is a partial template), and just overrides some blocks, all the other overrides which we are not changing, have to be included in the theme template because otherwise they will be missing in the page. I will quickly check this in my devstack and post an update.

Comment (lgp171188) [reply ecommerce/templates/edx/partials/_student_navbar.html:9]: I also checked the campus ecommerce theme repository and found the same sort of full template overrides as I have done here.

Comment (lgp171188) [reply ecommerce/templates/edx/partials/_student_navbar.html:9]: This is what the docstring in the `theming` module states

```
Theming aware template loaders, this loader will first look in template directories of current theme and then it will look at system template dirs.
```
`_student_navbar.html` extends `_base_navbar.html`. So when a template includes `_student_navbar.html`, the template loader will look for the template in the theme directories and if it is found there, theme version of the template will be used. If not the default version in the `ecommerce` repository will be used.

If we can extend the base version of the `_student_navbar.html` template in our theme `_student_navbar.html`, only then we can minimize the changes to just the blocks we are overriding. But this cannot be done because there is no way to refer to the default template directly because the template loader will pick the theme version every time.

Comment (lgp171188) [reply ecommerce/templates/edx/partials/_student_navbar.html:9]: One possible way to do this would be to override the templates which includes this template and make it include a different template file which extends the default template with this name. But that again will cause maintaining the whole template code for those overridden templates 😐

Comment (UmanShahzad) [reply ecommerce/templates/edx/partials/_student_navbar.html:9]: Ah I see, well that basically means ecommerce doesn't support the same kind of theming as the edx-platform, so yeah we have no choice yet but to either change how theming works to match edx-platform or just do full overrides.

File: ecommerce/static/sass/base/_variables.scss
Comment (UmanShahzad) [inline ecommerce/static/sass/base/_variables.scss:1]: @lgp171188 I noticed this file was missing in the diff in Gitlab?

Comment (UmanShahzad) [reply ecommerce/static/sass/base/_variables.scss:1]: Ok I saw it in `static/sass/partials/utilities/_variables.scss` instead. If this was required, that's fine with me.

**Review by UmanShahzad — APPROVED** [review]
:+1: this looks good. The only thing to worry about is getting the override in Gitlab correct now. We should not merge this until we get that one fixed, and we need to make sure we're using the theme in our deployment going forward. cc @kaizoku

**Comment by lgp171188** [comment]
@UmanShahzad is this good to merge now?

**Comment by UmanShahzad** [comment]
@lgp171188 Only if it's safe. Won't this break any new deployments that use this branch, unless we also add appropriate configuration to use the theme repo?

cc @kaizoku

**Comment by lgp171188** [comment]
@UmanShahzad, let's wait for confirmation from @josh. I will mention in the HMS hawthorn upgrade ticket that this PR is not merged needs to be merged after the configuration updates are done.

**Pull request closed without merging**
